File: solutions/python/isbn-verifier/1/isbn_verifier.py
def is_valid(isbn):
    # check by numbers of characters
    if not len(isbn) in (10, 13):
        return False
    # normalize
    new_isbn = isbn.replace("-", "").upper()
    # check by numbers of characters
    len_isbn = len(new_isbn)
    if not len_isbn in (9, 10):
        return False
    # add verification digit
    last_digit = "0"
    if len_isbn == 9:
        new_isbn += last_digit
    sumatory = 0
    for i, character in enumerate(reversed(new_isbn), start=1):
        if character == "X" and i == 1:
            sumatory += 10 
            continue
        if not character.isdigit():
            return False
        sumatory += int(character) * i
    return sumatory % 11 == 0

File: solutions/java/isbn-verifier/1/src/main/java/IsbnVerifier.java
import java.lang.String;


class IsbnVerifier {

    boolean isValid(String stringToVerify) {
        
        int n = stringToVerify.length();
        if (n < 10 || n > 13) return false;

        String isbn = stringToVerify.replace("-", "");
        n = isbn.length();
        if (n < 9 || n > 10) return false;

        if (n == 9) {
            n = 10; 
            isbn = isbn + "0";
        }

        int sum = 0;
        char [] charArrray =  isbn.toCharArray();
        for (int i = 1; i <= n; ++i) {
            char ch = charArrray[n - i];
            if (ch == 'X' && i == 1) {
                sum += 10;
                continue;
            }
            if (!Character.isDigit(ch))return false;
            sum += (Character.getNumericValue(ch) * i);
        }
        return sum % 11 == 0;
    }

}
